### PR TITLE
Better endpoint naming, http code + better /move endpoint + add tags

### LINKED ISSUE #592: Bad naming in endpoints + wrong http codes
Rename properties in all endpoints : 

- `content_type_slug` => "content_type"
- `sub_content_type_slug` => `sub_content_types` (in plural form as it's a list of values)
- `display_name` => `public_name`
- `"profile": { "slug": "managers" }` => `"profile": "managers"`
- `"global_status": "Open"` => values are always lowercase (so the example must be lowercase too)
- `id` => `workspace_id` pour les workspace, `content_id` pour les contenus
- `status_slug` => `status`
- `role_slug` => `role`

Wrong HTTP code and content : 

`/archive`, `/delete`, `/unarchive`, `/undelete` => return 204 with no body

Wrong datetime format : 

documented : "2018-06-18T21:11:16.876Z"
expected : "2018-06-18T21:11:16Z"

### LINKED ISSUE #593: wrong `/move` structure
Current API is the following one : 

```
PUT
{
  "new_parent_id": 42
}
```

returning an empty result

Expected : 

```
PUT
{
  "parent_id": 42,
  "workspace_id": 23
}
```

returning the content object with up-to-date information.

The reason for these renaming : 

- the body with parent_id/workspace_id is "apply-able" on a content object (same property names)
- the returned structure should always include updated data.

### LINKED ISSUE #596: Move endpoint should return 200 with content instead of 204
move endpoint should return updated_content.

### PR DESCRIPTION
**Pull request by inkhey**
contains also #86 and #87 

closes https://github.com/tracim/tracim/issues/596
closes tracim/tracim#592
Closes tracim/tracim#593

+ Add simple tags for endpoints.

### CODE REVIEW

File: tracim/exceptions.py [hunk 1]
```diff
@@ -111,3 +115,7 @@ class UserDoesNotExist(TracimException):
 
 class UserNotFoundInTracimRequest(TracimException):
     pass
+
+
+class NotSameWorkspace(TracimException):
```
Comment (lebouquetin) [inline tracim/exceptions.py]: rename this to `WorkspacesDoNotMatch`

File: tracim/lib/utils/authorization.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
     JSONDecodeError = ValueError
 
 from tracim.exceptions import InsufficientUserWorkspaceRole, \
-    InsufficientUserProfile
+    InsufficientUserProfile, WorkspaceNotFoundInTracimRequest
```
Comment (lebouquetin) [inline tracim/lib/utils/authorization.py]: one import per line

File: tracim/lib/utils/authorization.py [hunk 3]
```diff
@@ -101,3 +101,28 @@ def wrapper(self, context, request: 'TracimRequest'):
...
+    UserRoleInWorkspace.CONTRIBUTOR or UserRoleInWorkspace.READER
+    :return: decorator
+    """
+    def decorator(func):
+
+        def wrapper(self, context, request: 'TracimRequest'):
+            user = request.current_user
+            try:
+                workspace = request.candidate_workspace
+            except WorkspaceNotFoundInTracimRequest:
```
Comment (lebouquetin) [inline tracim/lib/utils/authorization.py]: Make new workspace id mandatory in API, so it simplifies our code

File: tracim/lib/utils/authorization.py [hunk 4]
```diff
@@ -101,3 +101,28 @@ def wrapper(self, context, request: 'TracimRequest'):
...
     return decorator
+
+
+def require_candidate_workspace_role(minimal_required_role: int):
+    """
+    Decorator for view to restrict access of tracim request if role
+    is not high enough. Do nothing is candidate_workspace_role is not found.
+    :param minimal_required_role: value from UserInWorkspace Object like
+    UserRoleInWorkspace.CONTRIBUTOR or UserRoleInWorkspace.READER
+    :return: decorator
```
Comment (lebouquetin) [inline tracim/lib/utils/authorization.py]: Rewrite 

```
+    Decorator for view to restrict access of tracim request if role
+    is not high enough. Do nothing is candidate_workspace_role is not found.
+    :param minimal_required_role: value from UserInWorkspace Object like
+    UserRoleInWorkspace.CONTRIBUTOR or UserRoleInWorkspace.READER
+    :return: decorator
```

to

```
+    Restricts access to endpoint to minimal role or raise an exception.
+    :param minimal_required_role: value from UserInWorkspace Object like
+    UserRoleInWorkspace.CONTRIBUTOR or UserRoleInWorkspace.READER
+    :return: decorator
```

File: tracim/lib/utils/request.py [hunk 5]
```diff
@@ -2,7 +2,7 @@
 from pyramid.request import Request
 from sqlalchemy.orm.exc import NoResultFound
 
-from tracim.exceptions import NotAuthenticated
+from tracim.exceptions import NotAuthenticated, WorkspaceNotFoundInTracimRequest
```
Comment (lebouquetin) [inline tracim/lib/utils/request.py]: one import per line

File: tracim/lib/utils/request.py [hunk 6]
```diff
@@ -102,6 +105,21 @@ def candidate_user(self) -> User:
             self.candidate_user = self._get_candidate_user(self)
         return self._candidate_user
 
+    @property
+    def candidate_workspace(self) -> Workspace:
+        """
+        Get user from headers/body request. This user is not
```
Comment (lebouquetin) [inline tracim/lib/utils/request.py]: I believe you copied/pasted the code. It's not about getting user but getting workspace

File: tracim/lib/utils/request.py [hunk 7]
```diff
@@ -56,7 +59,7 @@ def current_workspace(self) -> Workspace:
         :return: Workspace of the request
         """
         if self._current_workspace is None:
-            self.current_workspace = self._get_workspace(self.current_user, self)
+            self.current_workspace = self._get_current_workspace(self.current_user, self)
```
Comment (lebouquetin) [inline tracim/lib/utils/request.py]: missing the first underscore in `self.current_workspace` ?

File: tracim/lib/utils/ttest [hunk 8]
```diff
@@ -0,0 +1,341 @@
+
+definitions:
```
Comment (lebouquetin) [inline tracim/lib/utils/ttest]: this file is commit by error?

**Review by lebouquetin — CHANGES_REQUESTED** [review]

File: tracim/lib/utils/ttest (hunk 8 above)
Comment (inkhey) [reply tracim/lib/utils/ttest]: yes.

Comment (inkhey) [reply tracim/lib/utils/ttest]: Deleted.

File: tracim/lib/utils/request.py (hunk 7 above)
Comment (inkhey) [reply tracim/lib/utils/request.py]: Yes. Done.

File: tracim/lib/utils/request.py (hunk 6 above)
Comment (inkhey) [reply tracim/lib/utils/request.py]: Comment updated.

File: tracim/lib/utils/request.py (hunk 5 above)
Comment (inkhey) [reply tracim/lib/utils/request.py]: Done.

File: tracim/lib/utils/authorization.py (hunk 2 above)
Comment (inkhey) [reply tracim/lib/utils/authorization.py]: Done.

File: tracim/exceptions.py (hunk 1 above)
Comment (inkhey) [reply tracim/exceptions.py]: Done.

File: tracim/lib/utils/authorization.py (hunk 4 above)
Comment (inkhey) [reply tracim/lib/utils/authorization.py]: Done.

File: tracim/lib/utils/authorization.py (hunk 3 above)
Comment (inkhey) [reply tracim/lib/utils/authorization.py]: Done.

File: tracim/views/core_api/workspace_controller.py [hunk 9]
```diff
@@ -145,21 +151,24 @@ def create_generic_empty_content(
...
-            content_type=creation_data.content_type_slug,
+            content_type=creation_data.content_type,
             workspace=request.current_workspace,
         )
         api.save(content, ActionDescription.CREATION)
         content = api.get_content_in_context(content)
         return content
 
-    @hapic.with_api_doc()
+    @hapic.with_api_doc(tags=[WORKSPACES_ENDPOINTS_TAG])
```
Comment (lebouquetin) [inline tracim/views/core_api/workspace_controller.py]: rename `WORKSPACES_ENDPOINTS_TAG` to `WORKSPACE_ENDPOINT_TAG` - this avoid multiple plural form which is source of typo

File: tracim/views/core_api/workspace_controller.py [hunk 10]
```diff
@@ -145,21 +151,24 @@ def create_generic_empty_content(
...
     @hapic.handle_exception(InsufficientUserProfile, HTTPStatus.FORBIDDEN)
     @hapic.handle_exception(WorkspaceNotFound, HTTPStatus.FORBIDDEN)
+    @hapic.handle_exception(InsufficientUserWorkspaceRole, HTTPStatus.FORBIDDEN)
+    @hapic.handle_exception(WorkspacesDoNotMatch, HTTPStatus.BAD_REQUEST)
     @require_workspace_role(UserRoleInWorkspace.CONTRIBUTOR)
+    @require_candidate_workspace_role(UserRoleInWorkspace.CONTRIBUTOR)
     @hapic.input_path(WorkspaceAndContentIdPathSchema())
     @hapic.input_body(ContentMoveSchema())
-    @hapic.output_body(NoContentSchema())
+    @hapic.output_body(ContentDigestSchema())  # nopep8
```
Comment (lebouquetin) [inline tracim/views/core_api/workspace_controller.py]: why do yo uhave a nopep8 here?

**Review by lebouquetin — CHANGES_REQUESTED** [review]
still 2 minor changes before merge

File: tracim/views/core_api/workspace_controller.py (hunk 10 above)
Comment (inkhey) [reply tracim/views/core_api/workspace_controller.py]: probably bad copy/paste.

File: tracim/views/core_api/workspace_controller.py (hunk 9 above)
Comment (inkhey) [reply tracim/views/core_api/workspace_controller.py]: Done for Users too

**Pull request merged**